### PR TITLE
NATO/CSATcrate added optional arguments v2.0

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [ ] Bug
2. [x] Enhancement

### What have you changed and why?
Information:added optional arguments for NATO- CSATcrate
new NATO- CSATcrate arguments

[object, //required
WeapTypes, WeapNum, //opt
ItemTypes, ItemNum, //opt
AmmoTypes, AmmoNum, //opt
ExplosiveTypes, ExplosiveNum, //opt
AttachmentTypes, AttachmentNum, //opt
BackpackTypes, BackpackNum, //opt
HelmetTypes, HelmetNum, //opt
VestTypes, VestNum, //opt
DeviceTypes, DeviceNum //opt
] call A3A_fnc_NATOcrate;

example:
[cursorTarget,
1, 100,
0, 0,
0, 0,
0, 0,
0, 0,
0, 0,
0, 0,
0, 0,
0, 0
] call A3A_fnc_NATOcrate;

would spawn up to 100 of 1 type of weapon, and coresponding ammo

[cursorTarget] call A3A_fnc_NATOcrate; //would spawn default ammocrate

Why?
i needed this to specify what loot to add to crate in Salvage mission.
This change allows me to specify which type of loot and quantity i want to spawn in the crates. this alteration will make it so a a unique function dont have to be made every time we need to spawn loot that should not be at the level or as random as what is in the standard NATO/CSATcrate.

it does not affect any existing utilization of the script!

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the Mission in Singleplayer?
2. [ ] Have you loaded the Mission in a Dedicated Server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

********************************************************
Notes:

### CODE REVIEW

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 1]
```diff
@@ -161,134 +160,152 @@ else
...
-			[4, format ["Spawning %1 magazines for %2", _magAmount, _loot], _filename] call A3A_fnc_log;
-			_crate addMagazineCargoGlobal [selectrandom _magazines, _magAmount];
-			[4, format ["Spawning %1 of %2", _amount, _loot], _filename] call A3A_fnc_log;
+		if (isNil "_loot") then {
+			[3, "No Weapons Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else
+		{
+			[4, format ["Adding weapon: %1", _loot], _filename] call A3A_fnc_log;
+			_amount = if (isNil "_crateWepNumMax") then {crateWepNumMax call _fnc_pickAmount;} else {_crateWepNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateWepNumMax will never be nil, as it's defaulted. You'd need to remove the defaults to make this work. 

Also, it's not really _crateWepNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that. Something like "_createWepNumOverride"

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 2]
```diff
@@ -161,134 +160,152 @@ else
...
+		_available = (lootItem - _unlocks - itemCargo _crate);
+		[4, format ["Breakdown: %1, %2, %3", lootItem, _unlocks, itemCargo _crate], _filename] call A3A_fnc_log;
+		[4, format ["Items available: %1", _available], _filename] call A3A_fnc_log;
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Items Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			[4, format ["Item chosen: %1", _loot], _filename] call A3A_fnc_log;
+			_amount = if (isNil "_crateItemNumMax") then { round random crateItemNumMax;} else {_crateItemNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateItemNumMax will never be nil, as it's defaulted. You'd need to remove the defaults to make this work.

Also, it's not really _crateItemNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 3]
```diff
@@ -161,134 +160,152 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateAmmoTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateAmmoTypeMax do {
+		_available = (lootMagazine - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Ammo Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateAmmoNumMax") then {crateAmmoNumMax call _fnc_pickAmount;} else {_crateAmmoNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateAmmoNumMax will never be nil, as it's defaulted. You'd need to remove the defaults to make this work.

Also, it's not really _crateAmmoNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 4]
```diff
@@ -161,134 +160,152 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateExplosiveTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateExplosiveTypeMax do {
+		_available = (lootExplosive - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Explosives Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateExplosiveNumMax") then { round random crateExplosiveNumMax;} else {_crateExplosiveNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateExplosiveNumMax will never be nil, as it's defaulted. You'd need to remove the defaults to make this work.

Also, it's not really _crateExplosiveNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 5]
```diff
@@ -161,134 +160,152 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateAttachmentTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateAttachmentTypeMax do {
+		_available = (lootAttachment - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Attachment Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateAttachmentNumMax") then { crateAttachmentNumMax  call _fnc_pickAmount;} else {_crateAttachmentNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateAttachmentNumMax will never be nil, as it's defaulted. You'd need to remove the defaults to make this work.

Also, it's not really _crateAttachmentNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 6]
```diff
@@ -161,134 +160,152 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateBackpackTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateBackpackTypeMax do {
+		_available = (lootBackpack - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Backpacks Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateBackpackNumMax") then {round random crateBackpackNumMax;} else {_crateBackpackNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateBackpackNumMax will never be nil, as it's defaulted. You'd need to remove the defaults to make this work.

Also, it's not really _crateBackpackNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 7]
```diff
@@ -161,134 +160,152 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateHelmetTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateHelmetTypeMax do {
+		_available = (lootHelmet - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Helmets Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateHelmetNumMax") then { round random crateHelmetNumMax;} else {_crateHelmetNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateHelmetNumMax will never be nil, as it's defaulted. You'd need to remove the defaults to make this work.

Also, it's not really _crateHelmetNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 8]
```diff
@@ -161,134 +160,152 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateVestTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateVestTypeMax do {
+		_available = (lootVest - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Vests Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateVestNumMax") then { round random crateVestNumMax;} else {_crateVestNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateVestNumMaxwill never be nil, as it's defaulted. You'd need to remove the defaults to make this work.

Also, it's not really _crateVestNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf [hunk 9]
```diff
@@ -161,134 +160,152 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateDeviceTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateDeviceTypeMax do {
+		_available = (lootDevice - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Device Bags Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateDeviceNumMax") then { round random crateDeviceNumMax;} else {_crateDeviceNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: _crateDeviceNumMax never be nil, as it's defaulted. You'd need to remove the defaults to make this work.

Also, it's not really _crateDeviceNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 10]
```diff
@@ -189,139 +188,157 @@ else
...
-				floor random 3;
-			} else {
-				floor random [1,6,1]
+		if (isNil "_loot") then {
+			[3, "No Weapons Left in Loot List Or Pick Random Failed", _filename] call A3A_fnc_log;
+		}
+		else 
+		{
+			[4, format ["Adding weapon: %1", _loot], _filename] call A3A_fnc_log;
+			_amount = if (isNil "_crateWepNumMax") then {crateWepNumMax call _fnc_pickAmount;} else {_crateWepNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateWepNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 11]
```diff
@@ -189,139 +188,157 @@ else
...
+		_available = (lootItem - _unlocks - itemCargo _crate);
+		[4, format ["Breakdown: %1, %2, %3", lootItem, _unlocks, itemCargo _crate], _filename] call A3A_fnc_log;
+		[4, format ["Items available: %1", _available], _filename] call A3A_fnc_log;
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Items Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			[4, format ["Item chosen: %1", _loot], _filename] call A3A_fnc_log;
+			_amount = if (isNil "_crateItemNumMax") then { round random crateItemNumMax;} else {_crateItemNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateItemNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 12]
```diff
@@ -189,139 +188,157 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateAmmoTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateAmmoTypeMax do {
+		_available = (lootMagazine - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Ammo Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateAmmoNumMax") then {crateAmmoNumMax call _fnc_pickAmount;} else {_crateAmmoNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateAmmoNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 13]
```diff
@@ -189,139 +188,157 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateExplosiveTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateExplosiveTypeMax do {
+		_available = (lootExplosive - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Explosives Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateExplosiveNumMax") then { round random crateExplosiveNumMax;} else {_crateExplosiveNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateExplosiveNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 14]
```diff
@@ -189,139 +188,157 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateAttachmentTypeMax != 0) then {
+	for "_i" from 0 to (_crateAttachmentTypeMax call _fnc_pickNumberOfTypes) do {
+		_available = (lootAttachment - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Attachment Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateAttachmentNumMax") then { crateAttachmentNumMax  call _fnc_pickAmount;} else {_crateAttachmentNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateAttachmentNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 15]
```diff
@@ -189,139 +188,157 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateBackpackTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateBackpackTypeMax do {
+		_available = (lootBackpack - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Backpacks Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateBackpackNumMax") then {round random crateBackpackNumMax;} else {_crateBackpackNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateBackpackNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 16]
```diff
@@ -189,139 +188,157 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateHelmetTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateHelmetTypeMax do {
+		_available = (lootHelmet - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Helmets Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateHelmetNumMax") then { round random crateHelmetNumMax;} else {_crateHelmetNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateHelmetNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 17]
```diff
@@ -189,139 +188,157 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateVestTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateVestTypeMax do {
+		_available = (lootVest - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Vests Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateVestNumMax") then { round random crateVestNumMax;} else {_crateVestNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateVestNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf [hunk 18]
```diff
@@ -189,139 +188,157 @@ else
...
-		[4, format ["Spawning %2 of %3", _amount,_loot], _filename] call A3A_fnc_log;
+if (_crateDeviceTypeMax != 0) then {
+	for "_i" from 0 to floor random _crateDeviceTypeMax do {
+		_available = (lootDevice - _unlocks - itemCargo _crate);
+		_loot = selectRandom _available;
+		if (isNil "_loot") then {
+			[3, "No Device Bags Left in Loot List", _filename] call A3A_fnc_log;
+		}
+		else {
+			_amount = if (isNil "_crateDeviceNumMax") then { round random crateDeviceNumMax;} else {_crateDeviceNumMax};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_NATOCrate.sqf]: It's not really _crateDeviceNumMax if you don't use random - it's a fixed number of items passed in as a parameter, which is quite different, and the naming ought to reflect that.

File: A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf (hunk 1 above)
Comment (HakonRydland) [reply A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: NumMax is not defaulted TypeMax is

Comment (HakonRydland) [reply A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: no longer defaulted in params in CSATcrate

Comment (HakonRydland) [reply A3-Antistasi/functions/Ammunition/fn_CSATCrate.sqf]: renamed to no longer say max: _crateXNumMax to _crateXNum.

**Pull request merged**